pkg/controller/stack/stack_controller.go

```diff
@@ -576,7 +576,29 @@ func reconcileActiveVersions(stackResource *kabanerov1alpha2.Stack, c client.Cli
 			stackResource.Spec.Versions[i] = curSpec
 
 			// Update the status of the Stack object to reflect the images used
-			newStackVersionStatus.Images = curSpec.Images
+			for _, img := range curSpec.Images {
+				newStackVersionStatus.Images = append(newStackVersionStatus.Images, kabanerov1alpha2.ImageStatus{Id: img.Id, Image: img.Image})
+			}
+
+			// Retrieve stack image version activation digests. As such, it should only be done once during activation.
+			// If there is an error during first retrieval, a subsequent successful retry may set the current digest and
+			// not the activation digest. More precisely, the digest may not necessarily be the initial activation digest
+			// because we allow stack activation despite there being a failure when retrieving the digest and the
+			// image/digest may have changed before the next successful retry.
+			for j, image := range newStackVersionStatus.Images {
+				if len(image.Digest.Activation) == 0 {
```

Comment:
Isn't the length of `image.Digest.Activation` always going to be zero, because you just copied `newStackVersionStatus.Images` from `curSpec.Images` and `curSpec.Images` doesn't have any digest information?  It seems like the logic here needs to also look at the current status to see if we already had a digest for that version of the stack (and image).